api.c

```diff
@@ -160,6 +160,27 @@ void tcmu_seek_in_iovec(struct iovec *iovec, size_t count)
 	}
 }
 
+/*
+ * Consume an iovec. Count must not exceed the total iovec[] size.
+ * iove count should be updated.
+ */
+void tcmu_seek_in_cmd_iovec(struct tcmulib_cmd *cmd, size_t count)
+{
+	struct iovec *iovec = cmd->iovec;
+	while (count) {
+		if (count >= iovec->iov_len) {
+			count -= iovec->iov_len;
+			iovec->iov_len = 0;
+			iovec++;
+			cmd->iov_cnt--;
+		} else {
+			iovec->iov_base += count;
+			iovec->iov_len -= count;
+			count = 0;
+		}
+	}
+}
```

Comment:
For this patch could we just do the attached so we do not duplicate that loop.

[api-seek.txt](https://github.com/open-iscsi/tcmu-runner/files/1790648/api-seek.txt)